MantidPlot/src/ProjectRecoveryGUIs/ProjectRecoveryView.cpp

```diff
@@ -0,0 +1,93 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "ProjectRecoveryView.h"
+#include "ApplicationWindow.h"
+#include "MantidKernel/UsageService.h"
+#include "MultiTabScriptInterpreter.h"
+#include "Script.h"
+#include "ScriptFileInterpreter.h"
+#include "ScriptingWindow.h"
+#include "ui_ProjectRecoveryWidget.h"
+#include <boost/smart_ptr/make_shared.hpp>
+
+ProjectRecoveryView::ProjectRecoveryView(QWidget *parent,
+                                         ProjectRecoveryPresenter *presenter)
+    : QDialog(parent), ui(new Ui::ProjectRecoveryWidget),
+      m_presenter(presenter) {
+  ui->setupUi(this);
+  ui->tableWidget->horizontalHeader()->setResizeMode(QHeaderView::Stretch);
+  ui->tableWidget->verticalHeader()->setResizeMode(QHeaderView::Stretch);
+  ui->progressBar->setMinimum(0);
+  // Set the table information
+  addDataToTable(ui);
+  Mantid::Kernel::UsageService::Instance().registerFeatureUsage(
+      "Interface", "ProjectRecoveryWindow", true);
+}
+
+ProjectRecoveryView::~ProjectRecoveryView() { delete ui; }
```

Comment:
Why not use `std::unique_ptr`?

Reply:
It's a weird UI object thing and I originally tried to use a std::unique_ptr but it didn't like that went attempting to compile.

Reply:
I have been able to remove them as they had no actual use and were only there to satisfy the cppcheck for the rule of 3/5 when it comes to constructors and copy assignment, due to removing the destructor.